src/components/GradeLevelFilter.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Button, Popover } from '@material-ui/core';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+
+const useStyles = makeStyles((theme) => ({
+  buttonStyle: {
+    margin: theme.spacing(1),
+    height: theme.spacing(4),
+  },
+  popOverWindow: {
+    minWidth: '200px',
+  },
+}));
+
+const gradeList = {
+  12: {
+    value: 'Highschool Senior',
+    isChecked: false,
+  },
+  13: {
+    value: 'College Freshman',
+    isChecked: false,
+  },
+  14: {
+    value: 'College Sophomore',
+    isChecked: false,
+  },
+  15: {
+    value: 'College Junior',
+    isChecked: false,
+  },
+  16: {
+    value: 'College Senior',
+    isChecked: false,
+  },
+};
+
+export default function GradeLevelFilter() {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState(null);
+  const handleClick = (event) => setAnchorEl(event.currentTarget);
+
+  function updateGradeSelection(value) {
+    gradeList[value].isChecked = !gradeList[value].isChecked;
```

Comment:
This is potentially unsafe as this is not a true constant variable. It's being modified by potentially multiple instances of `GradeLevelFilter`.

Instead of this I recommend using [local state](https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class). Perhaps using `useState`.

Reply:
Hm, if we decide to go with a `set` then local state would not longer be needed since in the updateGradeSelection function would do something as follows:
```javascript
function updateGradeSelection(value) {
  if (gradeSet.has(value)) { // case when the grade selected is already in the set should be removed
    gradeSet.delete(value);
    return;
  }
  // happy case when the value just gets added to the set
  gradeSet.add(value); 
}
```

Reply:
You need to consider the case where `GradeLevelFilter` is used more than once. There's no guarantee of that. `GradeLevelFilter` could be re-rendered, which creates another instance and another callback and if multiple callbacks access the same global variable and tamper with it they could mess up the state.